oarepo_vocabularies/services/service.py

```diff
@@ -78,6 +81,71 @@ def _vocabulary_statistics(self):
 
 
 class VocabulariesService(InvenioVocabulariesService):
+    @functools.lru_cache()
+    def get_specialized_service(self, _type):
+        service_name = current_app.config.get("OAREPO_VOCABULARIES_SPECIALIZED_SERVICES", {}).get(_type)
+        if service_name:
+            return current_service_registry.get(service_name)
+
+    def search(
+        self, identity, params=None, search_preference=None, type=None, **kwargs
+    ):
+        specialized_service = self.get_specialized_service(type)
+        if specialized_service:
+            return specialized_service.search(
+                identity=identity, params=params, search_preference=search_preference, **kwargs)
+        return super().search(identity=identity, params=params,
+                              search_preference=search_preference, type=type, **kwargs)
+
+    def search_many(self, identity, params):
+        # we are skipping Invenio vocabularies service here and calling
+        # explicitly its parent class. The reason is that invenio vocabs
+        # always filter the search by a single vocabulary type. The search_many
+        # use case is an optimization where we want to fetch multiple items
+        # from multiple vocabulary types in a single query.
+        return super(InvenioVocabulariesService, self).search(identity, params)
+
+    def read_all(self, identity, fields, type, cache=True, extra_filter="", **kwargs):
+        specialized_service = self.get_specialized_service(type)
+        if specialized_service:
+            return specialized_service.read_all(identity=identity,
+                                                fields=fields,
+                                                cache=cache,
+                                                extra_filter=extra_filter,
+                                                **kwargs)
+        return super().read_all(identity=identity, fields=fields, type=type,
+                                cache=cache, extra_filter=extra_filter, **kwargs)
+
+    def read_many(self, identity, type, ids, fields=None, **kwargs):
+        """Search for records matching the querystring filtered by ids."""
+        specialized_service = self.get_specialized_service(type)
+        if specialized_service:
+            return specialized_service.read_many(identity=identity, ids=ids, fields=fields, **kwargs)
+        return super().read_many(identity=identity, type=type, ids=ids, fields=fields, **kwargs)
+
+    @unit_of_work()
+    def create(self, identity, data, uow=None, expand=False, **kwargs):
+        specialized_service = self.get_specialized_service(data.get("type"))
+        if specialized_service:
+            data.pop("type")
```

Comment:
Shouldn't we pop every key not matching target service schema here (like `props`, `title`)?

Reply:
no, the "type" in the payload is there just to be able to select the vocabulary, the rest of the items must match the vocabulary schema.
That's why only the type is popped, the rest must be according to the schema.